File: src/main/java/itmo/is/project/model/module/storage/StorageModule.java
package itmo.is.project.model.module.storage;

import itmo.is.project.model.module.Module;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "storage_module")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class StorageModule extends Module<StorageModuleBlueprint> {

    @OneToMany(mappedBy = "storageModule", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    private List<StoredResource> storedResources;
}

File: src/main/kotlin/itmo/isproject/storage/model/StorageModule.kt
package itmo.isproject.storage.model

import itmo.isproject.shared.module.model.Module
import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "storage_module")
class StorageModule(

    @OneToMany(mappedBy = "storageModule", fetch = FetchType.LAZY, cascade = [CascadeType.ALL], orphanRemoval = true)
    var storedResources: MutableList<StoredResource> = ArrayList(),

    blueprint: StorageModuleBlueprint? = null

) : Module<StorageModuleBlueprint>(blueprint)
